FordLabs/retroquest
# Partial React Migration

## Overview
This is the migration work done so far converting the front end from angular to react.

### Demo
**Starting App:**
- Now to start the app with the angular front end, run: `start`
- To start the app in react (what is converted to react thus far), run `start:react`

**Building App:**
- To build the app in angular, run `yarn build-prod`
- To build the app in react (what is converted to react thus far), run `yarn build-prod:react`

**Starting Storybook to view react components:**
- Run `yarn storybook` to start up the storybook that all the migrated react components are displayed in.
- Run `build-storybook` to build said storybook.

### Notes
What has been migrated so far:
- Most components have been migrated and a storybook has been created to display them
- The login page has been fully migrated to react

## Testing Instructions
1) Start the backend, and then start the front end in angular by running `start` and then run `start cypress` to ensure all cypress tests pass for the angular version of the app.
2) Run through using the angular app. Ensure everything works as before.
3) Start the backend, and then start the front end in react by running `start:react` and then run `start cypress` to ensure all cypress tests pass for the angular version of the app.
4) Run through using the react app. Ensure everything works as before.
